shared/wallets/routes.js

```diff
@@ -167,21 +167,28 @@ class WalletsSubNav extends React.PureComponent<any> {
   }
 }
 
-const WalletsSubNavigator = createNavigator(
-  WalletsSubNav,
-  StackRouter(Shim.shim(walletsSubRoutes), {initialRouteName: 'wallet'}),
-  {}
-)
-
-WalletsSubNavigator.navigationOptions = {
-  header: undefined,
-  headerTitle: 'Wallet',
-  title: 'Wallet',
-}
-
 export const newRoutes = {
   'tabs.walletsTab': {
-    getScreen: () => isMobile ? require('./wallet/container').default : WalletsSubNavigator, upgraded: true,
+    getScreen: () => {
+      if (isMobile) {
+        return require('./wallet/container').default
+      } else {
+        const WalletsSubNavigator = createNavigator(
```

Comment:
Creating a new navigator in here strikes me as fishy - how often does react navigation call this?

Reply:
Ah, looks like only once per launch: added a console.warn() and it only fired once.